pyvista/utilities/helpers.py

```diff
@@ -216,38 +215,82 @@ def get_array(mesh, name, preference='cell', info=False, err=False):
     preference = parse_field_choice(preference)
     if np.sum([parr is not None, carr is not None, farr is not None]) > 1:
         if preference == FieldAssociation.CELL:
-            if info:
-                return carr, FieldAssociation.CELL
-            else:
-                return carr
+            return carr
         elif preference == FieldAssociation.POINT:
-            if info:
-                return parr, FieldAssociation.POINT
-            else:
-                return parr
+            return parr
         elif preference == FieldAssociation.NONE:
-            if info:
-                return farr, FieldAssociation.NONE
-            else:
-                return farr
+            return farr
         else:
             raise ValueError(f'Data field ({preference}) not supported.')
-    arr = None
-    field = None
+
+    if parr is not None:
+        return parr
+    elif carr is not None:
+        return carr
+    elif farr is not None:
+        return farr
+    elif err:
+        raise KeyError(f'Data array ({name}) not present in this dataset.')
+    return None
+
+
+def get_array_association(mesh, name, preference='cell', err=False) -> FieldAssociation:
+    """Return the array association.
+
+    Parameters
+    ----------
+    mesh : Dataset
+        Dataset to get the array association from.
+
+    name : str
+        The name of the array.
+
+    preference : str, optional
+        When scalars is specified, this is the preferred array type to
+        search for in the dataset.  Must be either ``'point'``,
+        ``'cell'``, or ``'field'``
+
+    err : bool, optional
+        Boolean to control whether to throw an error if array is not present.
+
+    Returns
+    -------
+    :class:`pyvista.FieldAssociation`
+        Association of the array
+
+    """
+    if isinstance(mesh, _vtk.vtkTable):
+        arr = row_array(mesh, name)
+        if arr is None and err:
+            raise KeyError(f'Data array ({name}) not present in this dataset.')
+        return FieldAssociation.ROW
+
+    # with multiple arrays, return the array preference
+    parr = point_array(mesh, name)
+    carr = cell_array(mesh, name)
+    farr = field_array(mesh, name)
+    preference = parse_field_choice(preference)
+    if np.sum([parr is not None, carr is not None, farr is not None]) > 1:
```

Comment:
Chose the last.  There's no reason not to use `any`

Reply:
Agreed, stdlib `any` will also short-circuit (just a style concern, not really a performance one). For extra bikeshed points, consider using a generator expression instead of a list comprehension :)

Reply:
Turns out this has to be `sum` for it to work properly.  Had some weird errors and it turns out it's because we need to check if there are *more than one* rather `any`.

Reply:
> For extra bikeshed points, consider using a generator expression instead of a list comprehension :)

@adeak, if you're going after points, #1593 has unlimited potential.